src/System.Net.Sockets/src/System/Net/Sockets/BaseOverlappedAsyncResult.Windows.cs

```diff
@@ -95,14 +95,17 @@ private static unsafe void CompletionPortCallback(uint errorCode, uint numBytes,
                     }
                     else
                     {
+                        bool addRefedNativeOverlapped = false;
                         try
                         {
+                            asyncResult._nativeOverlapped.DangerousAddRef(ref addRefedNativeOverlapped);
```

Comment:
Ok, I understand now, I think.  You're saying that using DangerousOverlappedPointer in SocketPal is okay because we can be sure Dispose hasn't been called there, but we can't here, because we are not sure there's not a path where we could have disposed by here.  Is that correct?  Seems reasonable.

That said, 
(1) We should add a comment explaining that we should probably get rid of SafeNativeOverlapped usage in BaseOverlappedAsyncResult eventually, but we need to analyze all the possible usage scenarios and make sure there's no weird paths
(2) In this particular case, I think we should just use the passed in nativeOverlapped instead of the one on the asyncResult.